src/node/audio_buffer_source.rs

```diff
@@ -0,0 +1,673 @@
+use crossbeam_channel::{Receiver, Sender};
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+
+use crate::audio_buffer::AudioBuffer;
+use crate::context::{AsBaseAudioContext, AudioContextRegistration, AudioParamId};
+use crate::param::{AudioParam, AudioParamOptions, AutomationRate};
+use crate::render::{AudioParamValues, AudioProcessor, AudioRenderQuantum};
+use crate::{AtomicF64, SampleRate, RENDER_QUANTUM_SIZE};
+
+use super::{AudioNode, ChannelConfig, ChannelConfigOptions};
+
+// attributes shared between the node and the renderer
+//
+// @note - kind of pack both Scheduler and Controller with an additionnal
+//  `duration` entry and `seek` renamed to `offset`
+// @note - this could probably be generated by a macro! (`Vec<{name, default<type>}>`)
+//  maybe another macro could also help for Getters and Setters on the node itself
+#[derive(Clone, Debug)]
+struct SharedAttributes {
```

Comment:
Is is a reimplementation of `src/control.rs`

Reply:
yup that is the @note

Reply:
yup that's true but with the additional `duration` attribute and more consistent naming (`seek` -> `offset`)